jina/executors/indexers/cache.py

```diff
@@ -24,33 +24,58 @@ def post_init(self):
 class DocIDCache(BaseCache):
     """Store doc ids in a int64 set and persistent it to a numpy array """
 
+    # FIXME revert data structure.
+    # TODO do we really need a dict-like object?
+    store = {}
+
     def __init__(self, index_filename: str = None, *args, **kwargs):
         if not index_filename:
             # create a new temp file if not exist
             index_filename = tempfile.NamedTemporaryFile(delete=False).name
         super().__init__(index_filename, *args, **kwargs)
 
     def add(self, doc_id: UniqueId, *args, **kwargs):
```

Comment:
how can I add something to the `hash -> doc_id` ?

Reply:
What do you mean?

Atm the API for `add` is 

(from the Driver)

```
            self.exec_fn(doc.id, **{DATA_FIELD: key})
```

where the `key` can be either the `doc.id` or the `content_hash`. I think this could be simplified somehow, but not sure yet, as I'm still unclear on the expected use cases. 

So atm the data structure is either

`docid --> docid` (redundant, could be simplified somehow)
or
`docid --> content_hash`.

Why would you need to add `content_hash` -> `doc_id`?

Reply:
For Incremental Indexing, one may want to guarantee that the document with that `content` is not there and so u would check if that `content_hash` has been seen.

Reply:
But that's a query, right?

you'd first need to create a DocIdCache with a Driver that use the field `content_hash` for checking.

Then you add docs.

Then you do `query([content_hash])`

Reply:
Is this what you mean? https://github.com/jina-ai/jina/blob/958a29d84a7e1939c0e603fd65dc145536e67c22/tests/unit/drivers/test_cache_driver.py#L148